### PR TITLE
feat: support cursed inscriptions

### PR DESCRIPTION
**Pull request by rafaelcr**
* Keep `prev_output` and `prev_offset` in the locations table so we can calculate gaps
* Add `/sats/:ordinal/inscriptions` endpoint to list all cursed inscriptions for a single sat
* Support new chainhook event `cursed_inscription_revealed`
* Save and expose `curse_type` on all endpoints

### CODE REVIEW

File: src/api/schemas.ts [hunk 1]
```diff
@@ -282,6 +281,7 @@ export const ApiStatusResponse = Type.Object(
     status: Type.String(),
     block_height: Type.Optional(Type.Integer()),
     max_inscription_number: Type.Optional(Type.Integer()),
+    max_cursed_inscription_number: Type.Optional(Type.Integer()),
```
Comment (lgalabru) [inline src/api/schemas.ts:284]: why having this upper bound?

Comment (rafaelcr) [reply src/api/schemas.ts:284]: This actually just reports the max cursed number we currently have in the DB when you visit the `/ordinals` root endpoint. It won't put a cap on incoming inscriptions.

**Comment by rafaelcr** [comment]
@lgalabru a few questions

> the inscriptions data are nested in a cursed_inscription_reveal field (vs inscription_reveal)

Does a `cursed_inscription_reveal` event have the exact same schema as `inscription_reveal`?

> you should always wipe cursed inscriptions when ingesting blessed inscription is an idempotent operation

Can you expand a bit on this? When should I wipe cursed inscriptions?

> there's also a new field available that you should probably keep: `curse_type`

Gotcha, should I return this data on cursed inscription responses too?

**Comment by lgalabru** [comment]
> Does a cursed_inscription_reveal event have the exact same schema as inscription_reveal?

Yep

> Can you expand a bit on this? When should I wipe cursed inscriptions?

We can leave this for a future iteration, we're resetting blessed and cursed inscriptions pretty regularly. At some point, we will be adding new curse types, and I think we will keep the blessed inscriptions (to avoid re-ingesting 10M inscriptions), and only reset the cursed inscriptions.

> Gotcha, should I return this data on cursed inscription responses too?

That'd be great for the explorer I think.

**Review by lgalabru — APPROVED** [review]
Looking great, thanks @rafaelcr!

**Pull request merged**